### PR TITLE
Adds ability to edit CheckMo extension Billing Address

### PR DESCRIPTION
**Pull request by supernova-at**
## Description

This PR adds the ability to modify your billing address when you pay with Check / Money Order.

Out of scope are:
1. The ability to change your selected payment method once selected, and
2. Billing Address form improvements

both of which I have created internal tickets to address.

## Related Issue

Closes PWA-1701.

#### Test scenario(s) for direct fix/feature

Before performing the following tests, be sure to enable the Check / Money Order extension on your local development storefront by adding the following lines to your `venia-concept` `package.json` `dependencies` section:

```json
    "dependencies": {
        "@magento/venia-sample-payments-checkmo": "link:../extensions/venia-sample-payments-checkmo"
    }
```

##### Billing Address Same
1. Add item to guest cart
2. Navigate to `/checkout` page
3. Fill in Shipping Information and Shipping Methods sections
4. Select "Check / Money Order" in Payment Information section
5. Keep "Billing Address same as Shipping" checkbox checked
6. Verify that "Review Order" button works
7. Verify that an "Edit" button appears for the Payment Information section

##### Billing Address Same - Edit
1. Click to "Edit" your Payment Information
2. Verify that "Billing Address same as Shipping" is checked

##### Switch to Alternate Billing Address
1. Uncheck the "Billing Address same as Shipping" checkbox
2. Fill in a billing address
3. Click "Update"
4. Verify that there are no errors presented and that the edit dialog is dismissed

##### Billing Address Different - Edit
1. Click to "Edit" your Payment Information
1. Verify that "Billing Address same as Shipping" is *not* checked and that your different billing address is pre-populated in the form

##### Switch back to Same Billing Address
1. Re-check the "Billing Address same as Shipping" checkbox
1. Click "Update"
4. Verify that there are no errors presented and that the edit dialog is dismissed

##### Place Order
1. Verify that you can successfully place an order with a Check / Money Order payment method when the billing address is the same as the shipping address
1. Verify that you can successfully place an order with a Check / Money Order payment method when the billing address is *not* the same as the shipping address

#### Test scenario(s) for any existing impacted features/areas
1. Authenticated user flow still works
2. Switching back and forth between "billing address same as shipping" and not does not result in errors
3. Credit Card / Braintree payment flow still works

#### Test scenario(s) for any Magento Backend Supported Configurations

The Magento Backend likely needs to enable Check / Money Order payments for the actual placement of orders to succeed.

#### Is Browser/Device testing needed?

No

#### Any ad-hoc/edge case scenarios that need to be considered?

No

## Checklist

-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.

### CODE REVIEW

File: packages/extensions/venia-sample-payments-checkmo/src/components/edit.js
Comment (supernova-at) [inline packages/extensions/venia-sample-payments-checkmo/src/components/edit.js:32]: Add `propTypes`

File: packages/extensions/venia-sample-payments-checkmo/src/components/summary.css
Comment (supernova-at) [inline packages/extensions/venia-sample-payments-checkmo/src/components/summary.css:74]: Remove unused classes

File: packages/extensions/venia-sample-payments-checkmo/src/components/summary.js
Comment (supernova-at) [inline packages/extensions/venia-sample-payments-checkmo/src/components/summary.js:51]: Use `FormattedMessage`

File: packages/extensions/venia-sample-payments-checkmo/src/components/summary.js
Comment (supernova-at) [inline packages/extensions/venia-sample-payments-checkmo/src/components/summary.js:58]: Add `propTypes` and doc blocks.

File: packages/peregrine/lib/talons/CheckoutPage/BillingAddress/useBillingAddress.js
Comment (supernova-at) [inline packages/peregrine/lib/talons/CheckoutPage/BillingAddress/useBillingAddress.js:140]: I really only changed this to match the comment above.

File: packages/venia-concept/package.json
Comment (supernova-at) [inline packages/venia-concept/package.json:39]: Don't commit this change

File: yarn.lock
Comment (supernova-at) [inline yarn.lock:2928]: This change should be backed out too

File: packages/extensions/venia-sample-payments-checkmo/src/components/checkmo.js
Comment (larsroettig) [inline packages/extensions/venia-sample-payments-checkmo/src/components/checkmo.js:35]: Nice catcht

**Review by larsroettig — APPROVED** [review]

**Comment by larsroettig** [comment]
Looks good to send to QA would nice if you can cover new code with tests

**Comment by supernova-at** [comment]
## Coverage 

File                                                           | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
---------------------------------------------------------------|---------|----------|---------|---------|-------------------         
 venia-sample-payments-checkmo/src/components                  |     100 |      100 |     100 |     100 |                   
  checkmo.js                                                   |     100 |      100 |     100 |     100 |                   
  edit.js                                                      |     100 |      100 |     100 |     100 |                   
  summary.js                                                   |     100 |      100 |     100 |     100 |                   
 venia-sample-payments-checkmo/src/talons                      |     100 |      100 |     100 |     100 |                   
  checkmo.gql.js                                               |     100 |      100 |     100 |     100 |                   
  useCheckmo.js                                                |     100 |      100 |     100 |     100 |

**Comment by tjwiebell** [comment]
This is QA Passed ✅  but had one UX observation. First time I've seen multiple payment options, but the grid looks a little off when there are. Doesn't appear to have been introduced in your PR, so I'll create a bug and we can confirm with product if we should fix before release.

Screen Shot 2021-06-28 at 5 31 51 PM

**Review by tjwiebell — APPROVED** [review]
Re-approving after fixing merge conflict.

**Pull request merged**
